staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -136,22 +135,18 @@ func (f *structuredMergeManager) Update(liveObj, newObj runtime.Object, managed
 }
 
 // Apply implements Manager.
-func (f *structuredMergeManager) Apply(liveObj runtime.Object, patch []byte, managed Managed, manager string, force bool) (runtime.Object, Managed, error) {
-	patchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
-	if err := yaml.Unmarshal(patch, &patchObj.Object); err != nil {
-		return nil, nil, errors.NewBadRequest(fmt.Sprintf("error decoding YAML: %v", err))
-	}
-
+func (f *structuredMergeManager) Apply(liveObj, patchObj runtime.Object, managed Managed, manager string, force bool) (runtime.Object, Managed, error) {
 	// Check that the patch object has the same version as the live object
-	if patchObj.GetAPIVersion() != f.groupVersion.String() {
+	if patchObj.GetObjectKind().GroupVersionKind().GroupVersion() != f.groupVersion {
```

Comment:
why is this different?

Reply:
runtime.Object does not have GetAPIVersion() but the previous unstructured did.

Reply:
Thanks!